File: src/main/java/mnm/mods/tabbychat/ChatChannel.java
package mnm.mods.tabbychat;

import java.util.Objects;

public class ChatChannel extends AbstractChannel {

    private final String name;

    ChatChannel(String name) {
        super(name);
        this.name = name;
    }

    @Override
    public String getName() {
        return this.name;
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(name);
    }

    @Override
    public String toString() {
        return "#" + name;
    }

}

File: src/main/kotlin/mnm/mods/tabbychat/client/ChatChannel.kt
package mnm.mods.tabbychat.client

open class ChatChannel internal constructor(
        override val name: String) : AbstractChannel(name) {

    override val displayName: String
        get() = "#" + alias!!

    override fun hashCode(): Int {
        return toString().hashCode()
    }

    override fun toString(): String {
        return "#$name"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as ChatChannel

        if (name != other.name) return false

        return true
    }

}
